roslibrust/tests/ros1_xmlrpc.rs

```diff
@@ -94,26 +78,27 @@ mod tests {
             vec!["/verify_get_publications".into()],
         )
         .await;
+        log::info!("Got post advertise publications");
+
         assert_eq!(publications.len(), 1);
-        let (topic, topic_type) = publications.iter().nth(0).unwrap();
+        let (topic, topic_type) = publications
```

Comment:
Is this modification still necessary?

Reply:
Not sure... Potentially.

Generally I found some pretty horrible behavior with any kind of panic! not cleaning things up correctly. I'm not sure if the root problem was the DashMap and now we have good panic behavior or not.